src/Orchard.Web/Modules/Orchard.Setup/Services/SetupService.cs

```diff
@@ -174,7 +177,7 @@ public class SetupService : Component, ISetupService {
                 }
             }
 
-            // Set shell state to "Initializing" so that subsequent HTTP requests are responded to with "Service Unavailable" while Orchard is setting up.
+            // Reset shell state to "Initializing" so that subsequent HTTP requests are responded to with "Service Unavailable" while Orchard is setting up.

```

Comment:
On line 169 the state is set to `Running`. This seems flakey. What if a request comes in while the state is `Running`? Wouldn't it be better to modify the code in `CreateTenantData` so that it functions for both `Running` and `Initializing`?